implementations/rust/ockam/ockam_abac/src/policy.rs

```diff
@@ -57,20 +61,42 @@ impl PolicyAccessControl {
             environment: env,
         }
     }
+
+    pub async fn prepend_global_expression(&self, scoped_expression: Expr) -> Result<Expr> {
```

Comment:
I think the function name is misleading. Also, @etorreborre suggested a different way to implement such things, which I we should follow IMO. See `IdentitiesAttributes` as an example.

The structure is like this:
1. Have a simple SQL layer that only works with the storage - `PoliciesRepository` trait and `PolicySqlxDatabase` implementaiton
2. Have a layer on top, that can work with the storage, but also has business logic like verifying, aggregating data.  In this case it can be `Policies` struct,  which could have `Policies::get_policy` that will be given all the details about the resource, it would check for both specific and global policy in the storage, and return the Policy in effect (if any).

WDYT?

Reply:
Done